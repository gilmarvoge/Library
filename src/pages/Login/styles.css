#page-login {
  height:100vh;
  width: 100%;
  margin: 0;
  background: url("../../assets/login-background.svg");
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: bottom right;
}

#page-login form {
  margin: 0px auto;
  margin-top: 50px;
  margin-left: 20%;
  padding: 64px;
  max-width: 400px;
  background: #fff;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
}

#page-login form header {
  display: flex;
  justify-content: space-between;
}

#page-login form header h1 {
  font-size: 54px;
  vertical-align: middle;
}

#page-login form fieldset {
  margin-top: 64px;
  min-inline-size: auto;
  border: 0;
}

#page-login form .field {
  flex: 1;
  display: flex;
  flex-direction: column;
  margin-bottom: 24px;
}

#page-login form .field input[type="text"], #page-login form .field input[type="password"] {
  flex: 1;
  background: #f0f0f5;
  border-radius: 8px;
  border: 0;
  padding: 16px 24px;
  font-size: 16px;
  color: #6c6c80;
}

#page-login form .field input::placeholder {
  color: #a0a0b2;
}

#page-login form .field label {
  font-size: 14px;
  margin-bottom: 8px;
}

#page-login form button {
  height: 56px;
  background: var(--primary-color);
  border-radius: 8px;
  color: #fff;
  font-weight: bold;
  font-size: 16px;
  border: 0;
  align-items: center;
  margin-top: 40px;
  transition: background-color 0.2s;
  cursor: pointer;
  display: flex;
  outline: auto;
}

#page-login form button span {
  display: block;
  background: rgba(0, 0, 0, 0.08);
  width: 72px;
  height: 56px;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: background-color 0.2s;
}

#page-login form button span svg {
  color: #fff;
  width: 20px;
  height: 20px;
}

#page-login form button strong {
  flex: 1;
  text-align: center;
  color: #fff;
}

#page-login form button:hover {
  background: var(--hover-color);
}

@media (max-width: 500px) {
  #page-login form header h1 {
    font-size: 42px;
  }
  #page-login form p {
    font-size: 24px;
  }
}

