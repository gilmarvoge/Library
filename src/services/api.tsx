//API Fict√≠cia
const api = [
  {
    id: 1,
    user: "gilmar",
    password: "12345",
  },
  {
    id: 2,
    user: "soft",
    password: "012345",
  }
]
export default api;
