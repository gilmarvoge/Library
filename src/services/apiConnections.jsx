const apiUrl = 'http://localhost:3004';
    
export const urlUsers = `${apiUrl}/users`;
export const urlBooks = `${apiUrl}/books`;
export const urlRents = `${apiUrl}/books`;




