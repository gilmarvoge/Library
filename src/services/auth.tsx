
export const TOKEN_KEY = "token";

export const getUserIdStorage = () => localStorage.getItem(TOKEN_KEY);

export const isAuthenticated = () => localStorage.getItem(TOKEN_KEY) !== null;

export const login = (id: string) => {
  localStorage.setItem(TOKEN_KEY, id);
};

export function logoff() {
  localStorage.removeItem(TOKEN_KEY);
};
