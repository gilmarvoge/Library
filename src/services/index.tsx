export * from './api';
export * from './apiConnections';
export * from './auth';
export * from './mocks';