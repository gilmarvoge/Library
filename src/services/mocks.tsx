//Dados Fictcios
export const users = [
  {
    id: 1,
    user: "gilmar",
    password: "12345",
  },
  {
    id: 2,
    user: "soft",
    password: "012345",
  }
];

export const books = [
  {
    id: '7aa71482-d66e-4b47-a8b3-b7a08a3fb426',
    author: 'William E. Shotts, Jr.',
    title: 'The Linux Command Line',
    description: 'The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell.',
    image_url: 'http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api',
  },
  {
    id: '9f34a955-5636-49f3-b319-4bb54343c242',
    author: 'Artemij Fedosejev',
    title: 'React.js Essentials',
    description: 'Build maintainable and performant user interfaces for your web applications using React.js',
    image_url: 'http://books.google.com/books/content?id=Rhl1CgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
  },
  {
    id: '443bf367-5ff4-4d0c-92f1-8806458848d1',
    author: 'Artemij Fedosejev',
    title: 'Getting Started with React Native',
    description: 'Building web applications with maintainable and performant user interfaces is a challenge that many have faced for more than a decade, but no one has risen to this challenge quite like React.js. Today React.js is used by Facebook, Instagram, Khan Academy.',
    image_url: 'http://books.google.com/books/content?id=274fCwAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
  },
  {
    id: '036d85db-dcea-40b3-b120-0acf5277ccbf',
    author: 'Harmeet SinghMehul Bhatt',
    title: 'Learning Web Development with React and Bootstrap',
    description: 'Build maintainable and performant user interfaces for your web applications using React.js',
    image_url: 'http://books.google.com/books/content?id=r9JJvgAACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api',
  },
  {
    id: 'bbb67f51-4174-44f6-99d1-957eff015402',
    author: 'Vipul A MPrathamesh Sonpatki',
    title: 'ReactJS by Example - Building Modern Web Applications with React',
    description: 'Get up and running with ReactJS by developing five cutting-edge and responsive projects About This Book * Create pragmatic real-world applications while learning React and its modern developer tools ',
    image_url: 'http://books.google.com/books/content?id=vSLlCwAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
  },
];

export const rents = [
  {
    book_id: '443bf367-5ff4-4d0c-92f1-8806458848d1',
    user_id: '1',
  },
  {
    book_id: '7aa71482-d66e-4b47-a8b3-b7a08a3fb426',
    user_id: '2',
  },
  {
    book_id: '9f34a955-5636-49f3-b319-4bb54343c242',
    user_id: '2',
  },
];
