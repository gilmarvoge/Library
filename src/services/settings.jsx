export const userPath = '/users';
export const bookPath = '/books';
export const rentPath = '/rents';




