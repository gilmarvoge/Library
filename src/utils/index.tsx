export * from './mocks.tests.books';
export * from './mocks.tests.rents';
export * from './helpers'