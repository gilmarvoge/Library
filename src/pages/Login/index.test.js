import React from "react";
import Enzyme from 'enzyme';
import { users } from 'services/mocks';
import { Provider, LocalizeProvider } from 'react-redux';
import { Router } from 'react-router-dom';
import store from 'store';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history'
import Login from 'pages/Login';
import { render, fireEvent, screen } from '../../utils/test.utils';

// it('Renders the connected app with initialState', () => {
//   render(<Login />, { initialState: { user: '1' } })

//   expect(screen.getByText(/redux user/i)).toBeInTheDocument()
// })

describe("<Login />", () => {
  it("renders text input correctly", () => {
    // const historyMock  =jest.mock('react-router-dom', () => ({
    //   ...jest.requireActual('react-router-dom'),
    //   useHistory: () => ({
    //     push: jest.fn()
    //   })
    // }));
    const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
    const component  = renderer.create(   
        <Provider store={store}>    
        <Router history={historyMock}>  
          <Login />  
          </Router>         
        </Provider>
    )
       let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onLogin when button clicked", () => {
    const onSubmitMock = jest.fn();

    const component = Enzyme.mount(
      <Login onSubmit={onSubmitMock} />
    );

    component.find("input.user").simulate('change', { target: { value: users[0].user } })
    component.find("input.password").simulate('change', { target: { value: users[0].password } })
    component.find("form").simulate("submit");

    console.log("onClickMock.mock", onSubmitMock.mock)
    expect(onSubmitMock).toBeCalled()
  });
});