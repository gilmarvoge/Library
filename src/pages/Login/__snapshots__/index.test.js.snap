// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders text input correctly 1`] = `
<div
  id="page-login"
>
  <form
    onSubmit={[Function]}
  >
    <header>
      <img
        alt="Biblioteca"
        src="logo.svg"
      />
      <h1>
        Login
      </h1>
    </header>
    <fieldset>
      <div
        className="field"
      >
        <label
          htmlFor="name"
        >
          Usuario
        </label>
        <input
          id="user"
          name="user"
          type="text"
        />
      </div>
      <div
        className="field"
      >
        <label
          htmlFor="name"
        >
          Senha
        </label>
        <input
          id="password"
          name="password"
          type="password"
        />
      </div>
    </fieldset>
    <button
      type="submit"
    >
      <span>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
          />
          <polyline
            points="10 17 15 12 10 7"
          />
          <line
            x1="15"
            x2="3"
            y1="12"
            y2="12"
          />
        </svg>
      </span>
      <strong>
        Entrar
      </strong>
    </button>
  </form>
  
</div>
`;
